File: Implement two stacks in an array[14-July-2023]/Solution.java
//  >> JAVA CODE <<


class twoStacks
{
    int size,top1,top2;
    int[] arr;
    
    twoStacks()
    {
        this.size=100;
        this.arr=new int[size];
        this.top1=-1;
        this.top2=size;
    }
    
    
    void push1(int x)
    {
        if(top1>=top2) return;
        
        arr[++top1]=x;
    }
    //Function to push an integer into the stack2.
    void push2(int x)
    {
       if(top1>=top2) return;
       arr[--top2]=x;
    }
    //Function to remove an element from top of the stack1.
    int pop1()
    {
        if(top1==-1) return -1;
        return arr[top1--];
    }
    //Function to remove an element from top of the stack2.
    int pop2()
    {
        if(top2==size) return -1;
        return arr[top2++];
    }
}

File: Implement two stacks in an array[14-July-2023]/solution.cpp
//  >> PYTHON CODE <<

class twoStacks
{
    int *arr;
    int size;
    int top1, top2;
    public:
    
    twoStacks(int n=100)
    {
        size = n; 
        arr = new int[n]; 
        top1 = -1; 
        top2 = size;
    }
 
    //Function to push an integer into the stack1.
    void push1(int x)
    {
        if(top1>=top2) return;
        arr[++top1]=x;
    }
    
    //Function to push an integer into the stack2.
    void push2(int x)
    {
       if(top1>=top2) return;
        
       arr[--top2]=x;
    }
    
    //Function to remove an element from top of the stack1.
    int pop1()
    {
        return top1==-1?-1:arr[top1--];
    }
    
    //Function to remove an element from top of the stack2.
    int pop2()
    {
       return top2==size?-1:arr[top2++];
    }
};
